src/metadata.ts

```diff
@@ -3,83 +3,86 @@ import { z } from "zod";
 import { clone, mergeDeepRight } from "ramda";
 import { ProprietaryKind } from "./proprietary-schemas";
 
-export interface Metadata<T extends z.ZodTypeAny> {
+// @see https://github.com/colinhacks/zod/pull/3445/files#diff-d8a41ab20c64f92092513f153b78e7bbf5dc929c92909fad25e03d53c7f15bb7R21-R37
+export const metaSymbol = Symbol.for("express-zod-api");
+
+export interface Metadata {
   /**
    * @todo if the following PR merged, use native branding instead:
    * @link https://github.com/colinhacks/zod/pull/2860
    * */
   kind?: ProprietaryKind;
-  examples: z.input<T>[];
+  examples: unknown[];
 }
 
-export const metaProp = "expressZodApiMeta";
-
-type ExampleSetter<T extends z.ZodTypeAny> = (
-  example: z.input<T>,
-) => WithMeta<T>;
-
-type WithMeta<T extends z.ZodTypeAny> = T & {
-  _def: T["_def"] & Record<typeof metaProp, Metadata<T>>;
-  example: ExampleSetter<T>;
-};
+declare module "zod" {
+  interface ZodTypeDef {
+    [metaSymbol]: Metadata;
+  }
+  interface ZodType {
+    example: (example: this["_input"]) => this;
```

Comment:
this does not work for my case

Reply:
that way I'd have to alter prototypes of each class and I'll not do that

Reply:
the issue was the definition.
it has to be declared as method, not property having method.